rtos/mbed_boot.c

```diff
@@ -574,9 +576,14 @@ void pre_main(void)
     singleton_mutex_attr.cb_mem = &singleton_mutex_obj;
     singleton_mutex_id = osMutexNew(&singleton_mutex_attr);
 
+#if (__IAR_SYSTEMS_ICC__ >= 8)
+    __iar_Initlocks();
```

Comment:
we are having exit function here defined as well? As the docs say, we should invoke it (as initlocks in multithreaded env, added here.

Info: http://netstorage.iar.com/SuppDB/Public/UPDINFO/012120/arm/doc/infocenter/iccarm.ENU.html

(This info is useful to know, could be part of the commit message to include details what this initlocks is fixing).